### ISSUE TITLE
"My files" is slow to load

### ISSUE THREAD

**Issue by jcoyne** [projecthydra/sufia#845]
seeing: `Views: 4081.3ms`

**Comment by jcoyne** [comment]
Looks like it might be slow due to my/_index_partials/_list_files.html.erb.  It's possible that "to_model" is causing the slowdown?

**Comment by phebal** [comment]
I also saw a similar slowness, in my case it was caused by the `GenericFileHelper#render_collection_list` method.

**Comment by jcoyne** [comment]
@phebal is it any better for you if you use active-fedora from master?  I added some caching there yesterday.

**Comment by phebal** [comment]
Sorry, I tried but I it doesn't install with Sufia 4.3.1. I can't install Sufia branch `5.0.0-stable` (or the tagged rc1 release) because of bootstrap_form dependency problem (I'll try to fork it on Monday to fix this by hand and will update this ticket then):
```
Bundler could not find compatible versions for gem "bootstrap_form":
  In Gemfile:
    sufia (>= 0) ruby depends on
      hydra-editor (~> 0.3) ruby depends on
        bootstrap_form (~> 2.1.1) ruby

    sufia (>= 0) ruby depends on
      bootstrap_form (2.1.0)
```

And with Sufia 6 I get this trace (probably due to running Fedora 3?):
```
Ldp::HttpError (STATUS: 302 ...):
  ldp (0.2.1) lib/ldp/client/methods.rb:107:in `block in check_for_errors'
  ldp (0.2.1) lib/ldp/client/methods.rb:105:in `tap'
  ldp (0.2.1) lib/ldp/client/methods.rb:105:in `check_for_errors'
  ldp (0.2.1) lib/ldp/client/methods.rb:53:in `get'
  /home/phb010/.rvmruby (2.1.5) bundler/gems/active_fedora-6c6596547137/lib/active_fedora/caching_connection.rb:13:in `get'
  /home/phb010/.rvmruby (2.1.5) bundler/gems/active_fedora-6c6596547137/lib/active_fedora/fedora.rb:29:in `init_base_path'
  /home/phb010/.rvmruby (2.1.5) bundler/gems/active_fedora-6c6596547137/lib/active_fedora/fedora.rb:5:in `initialize'
  /home/phb010/.rvmruby (2.1.5) bundler/gems/active_fedora-6c6596547137/lib/active_fedora.rb:202:in `new'
  /home/phb010/.rvmruby (2.1.5) bundler/gems/active_fedora-6c6596547137/lib/active_fedora.rb:202:in `fedora'
  /home/phb010/.rvmruby (2.1.5) bundler/gems/active_fedora-6c6596547137/lib/active_fedora/ldp_cache.rb:40:in `reset_cache_settings'
  /home/phb010/.rvmruby (2.1.5) bundler/gems/active_fedora-6c6596547137/lib/active_fedora/ldp_cache.rb:33:in `rescue in call'
  /home/phb010/.rvmruby (2.1.5) bundler/gems/active_fedora-6c6596547137/lib/active_fedora/ldp_cache.rb:24:in `call'
  activerecord (4.1.6) lib/active_record/migration.rb:380:in `call'
  actionpack (4.1.6) lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
  activesupport (4.1.6) lib/active_support/callbacks.rb:82:in `run_callbacks'
  actionpack (4.1.6) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
  actionpack (4.1.6) lib/action_dispatch/middleware/reloader.rb:73:in `call'
  actionpack (4.1.6) lib/action_dispatch/middleware/remote_ip.rb:76:in `call'
  web-console (2.0.0) lib/action_dispatch/debug_exceptions.rb:18:in `middleware_call'
  web-console (2.0.0) lib/action_dispatch/debug_exceptions.rb:13:in `call'
  actionpack (4.1.6) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
  railties (4.1.6) lib/rails/rack/logger.rb:38:in `call_app'
  railties (4.1.6) lib/rails/rack/logger.rb:20:in `block in call'
  activesupport (4.1.6) lib/active_support/tagged_logging.rb:68:in `block in tagged'
  activesupport (4.1.6) lib/active_support/tagged_logging.rb:26:in `tagged'
  activesupport (4.1.6) lib/active_support/tagged_logging.rb:68:in `tagged'
  railties (4.1.6) lib/rails/rack/logger.rb:20:in `call'
  actionpack (4.1.6) lib/action_dispatch/middleware/request_id.rb:21:in `call'
  rack (1.5.2) lib/rack/methodoverride.rb:21:in `call'
  rack (1.5.2) lib/rack/runtime.rb:17:in `call'
  activesupport (4.1.6) lib/active_support/cache/strategy/local_cache_middleware.rb:26:in `call'
  rack (1.5.2) lib/rack/lock.rb:17:in `call'
  actionpack (4.1.6) lib/action_dispatch/middleware/static.rb:64:in `call'
  rack (1.5.2) lib/rack/sendfile.rb:112:in `call'
  railties (4.1.6) lib/rails/engine.rb:514:in `call'
  railties (4.1.6) lib/rails/application.rb:144:in `call'
  rack (1.5.2) lib/rack/content_length.rb:14:in `call'
  thin (1.6.3) lib/thin/connection.rb:86:in `block in pre_process'
  thin (1.6.3) lib/thin/connection.rb:84:in `catch'
  thin (1.6.3) lib/thin/connection.rb:84:in `pre_process'
  thin (1.6.3) lib/thin/connection.rb:53:in `process'
  thin (1.6.3) lib/thin/connection.rb:39:in `receive_data'
  eventmachine (1.0.4) lib/eventmachine.rb:187:in `run_machine'
  eventmachine (1.0.4) lib/eventmachine.rb:187:in `run'
  thin (1.6.3) lib/thin/backends/base.rb:73:in `start'
  thin (1.6.3) lib/thin/server.rb:162:in `start'
  rack (1.5.2) lib/rack/handler/thin.rb:16:in `run'
  rack (1.5.2) lib/rack/server.rb:264:in `start'
  railties (4.1.6) lib/rails/commands/server.rb:69:in `start'
  railties (4.1.6) lib/rails/commands/commands_tasks.rb:81:in `block in server'
  railties (4.1.6) lib/rails/commands/commands_tasks.rb:76:in `tap'
  railties (4.1.6) lib/rails/commands/commands_tasks.rb:76:in `server'
  railties (4.1.6) lib/rails/commands/commands_tasks.rb:40:in `run_command!'
  railties (4.1.6) lib/rails/commands.rb:17:in `<top (required)>'
  bin/rails:8:in `require'
  bin/rails:8:in `<top (required)>'
  spring (1.2.0) lib/spring/client/rails.rb:27:in `load'
  spring (1.2.0) lib/spring/client/rails.rb:27:in `call'
  spring (1.2.0) lib/spring/client/command.rb:7:in `call'
  spring (1.2.0) lib/spring/client.rb:26:in `run'
  spring (1.2.0) bin/spring:48:in `<top (required)>'
  spring (1.2.0) lib/spring/binstub.rb:11:in `load'
  spring (1.2.0) lib/spring/binstub.rb:11:in `<top (required)>'
  bin/spring:16:in `require'
  bin/spring:16:in `<top (required)>'
  bin/rails:3:in `load'
  bin/rails:3:in `<main>'


  Rendered /home/phb010/.rvm/gems/ruby-2.1.5/gems/web-console-2.0.0/lib/action_dispatch/templates/rescues/_source.erb (57.6ms)
  Rendered /home/phb010/.rvm/gems/ruby-2.1.5/gems/web-console-2.0.0/lib/action_dispatch/templates/rescues/_trace.html.erb (1.8ms)
  Rendered /home/phb010/.rvm/gems/ruby-2.1.5/gems/web-console-2.0.0/lib/action_dispatch/templates/rescues/_request_and_response.html.erb (0.6ms)
  Rendered /home/phb010/.rvm/gems/ruby-2.1.5/gems/web-console-2.0.0/lib/action_dispatch/templates/rescues/_web_console.html.erb (0.6ms)
  Rendered /home/phb010/.rvm/gems/ruby-2.1.5/gems/web-console-2.0.0/lib/action_dispatch/templates/rescues/diagnostics.html.erb within rescues/layout (71.7ms)
```

**Comment by jcoyne** [comment]
Yeah, my fix is only for Sufia 6.0.0.rc2 or master.

-Justin
